public/app/features/dashboard/folder_picker/folder_picker.ts

```diff
@@ -132,23 +132,26 @@ export class FolderPickerCtrl {
   }
 
   private loadInitialValue() {
-    if (this.initialFolderId && this.initialFolderId > 0) {
-      this.getOptions('').then(result => {
+    this.getOptions('').then(result => {
+      if (!_.isNil(this.initialFolderId)) {
+        // If initialFolderId is set, try to find it in result or return null
         this.folder = _.find(result, { value: this.initialFolderId });
         if (!this.folder) {
-          this.folder = { text: this.initialTitle, value: this.initialFolderId };
+          this.folder = { text: this.initialTitle, value: null };
```

Comment:
Wonder if not the only change needed in this file would be here - if initialFolderId not found, take the first from the result.

Reply:
There're 2 ways to go if initialFolderId was set but wasn't found:
1. Return null to explicitly notify user that folder wasn't found.
2. Just set to first available.
Not sure which is better. I just used 1.

Reply:
Would go for 2) first, if there are none, then use null.